bincommon/commonflags.go

```diff
@@ -132,6 +133,22 @@ func SharedMain(usage func(io.Writer, ...interface{})) {
 		"Size of the buffer (max data size) for the optimized http client in `kbytes`")
 	flag.BoolVar(&fhttp.CheckConnectionClosedHeader, "httpccch", fhttp.CheckConnectionClosedHeader,
 		"Check for Connection: Close Header")
+	// FlagResolveIPType indicates which IP types to resolve.
+	// With round robin resolution now the default, you are likely to get ipv6 which may not work if
+	// use both type (`ip`). In particular some test environments like the CI do have ipv6
+	// for localhost but fail to connect. So we made the default ip4 only.
+	dflag.Flag("resolve-ip-type", fnet.FlagResolveIPType)
```

Comment:
I'm trying to understand the re-use of the `fnet.FlagResolveMethod` into a new `dflag.Flag`. Partially because it's my first time checking these things.

I guess I'm wondering if there shouldn't be a more 'central' init place for all flags? Why we have some flags (?) "defined" in a different package that we can 're-init' as a dynamic flag in the 'main' part? Idk if my question makes sense but.. yea.

Reply:
Also sort of makes it easier I guess? Not having to comment both 'sides'

Reply:
So the reasoning here is the libraries define a config variable (using dflag.New) which includes the contract that they'll call .Get() to get the typed values, may setup validators, and the value can come from string using the same interface as flags.

Then a program, using the library, decides if they want to configure it as a dflag (or not)

In this case fortio is both a set of libraries, and the main user of them, so the flags are moved to bincommon for the most part

The comment on both sides is mostly that go doc wants exported variables commented (even though the 2nd argument of dflag.New is the comment/usage (which will be dumped on the flags UI and command line help) and when I copied it here I preserved it but yes it should only be needed once.

Reply:
In short in fnet/ and fhttp/ the "flags" aren't flags anymore they are dynamic config objects, that happen to be backed by dflags

This is to avoid that if you use a small part of fnet or fhttp you end up with your small cli `-help` showing a bunch of stuff you do not use. 

Ditto btw I got rid of log's many flag in favor of a Config object and a single optional flag

Reply:
> move parts of code comments into flag docs.

any example of which?

Reply:
> > move parts of code comments into flag docs.
> 
> any example of which?

```
  -resolve-ip-type type
    	Resolve type: ip4 for ipv4, ip6 for ipv6 only, use ip for both (default ip4)
```

vs 

```
// With round robin resolution now the default, you are likely to get ipv6 which may not work if
	// use both type (`ip`). In particular some test environments like the CI do have ipv6
	// for localhost but fail to connect. So we made the default ip4 only.
```

like the "code lacks" info about which options there are and the documentation "lacks" background information / reasoning. 

Having both in a single place is nice.

Reply:
So comments can be as long and as anecdotal or detailed as we want but the flag help text needs to be somewhat concise. I kinda wish the inverse: that if we don't document a config var / flag, the go doc would show the value / line for self doc

Reply:
we need what you're working on "fortio the book/tutorials" which can be informed from both flags help line and comments with more context added.

Reply:
noted in #266